src/pages/index.js

```diff
@@ -21,36 +37,121 @@ const imageModal = new ModalWithImage({
   modalSelector: "#view-image-modal",
 });
 
+const deleteModal = new ModalDelete({
+  modalSelector: "#delete-image-modal",
+  cardRemovalHandler: handleCardRemoval,
+  card: undefined,
+});
+
 // profile section
 
 const profileEl = new UserInfo({
   nameSelector: "#profile-name",
   descriptionSelector: "#profile-description",
+  avatarSelector: "#avatar",
 });
 
 // submission handlers
 
 function handleProfileEditSubmit(e, inputValues) {
   e.preventDefault();
+  consts.profileEditDiv.querySelector(".modal__button").innerText = "Saving...";
   const name = inputValues["Name"];
   const description = inputValues["Description"];
-  profileEl.setUserInfo(name, description);
-  profileEditValidator.disableSubmitButton();
+  return api
+    .updateProfileInfo(name, description)
+    .then(() => {
+      profileEl.setUserInfo(name, description);
+      formValidation[1].disableSubmitButton();
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+function handleAvatarEditSubmit(e, inputValues) {
+  e.preventDefault();
+  consts.avatarEditDiv.querySelector(".modal__button").innerText = "Saving...";
+  const avatar = inputValues["Avatar"];
+  return api
+    .updateAvatar(avatar)
+    .then(() => {
+      profileEl.setAvatar(avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
 }
 function handleAddCardSubmit(e, inputValues) {
   e.preventDefault();
+  consts.addCardModalDiv.querySelector(".modal__button").innerText =
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```